### PR TITLE
LPD-38634 Inhibit object test to get previous value by isolating tests

### PR DESCRIPTION
**Pull request by rijema**
Hi, y'all.

Motivation: 
[This test ](https://liferay.atlassian.net/browse/LPD-38634) was flaky in our routine affecting the suite result. I managed to reproduce the issue locally by setting a linear context where the mocked requests/responses values on BaseExportImportTestCase where instanced with the same value for all the test ( e.g: private MockLiferayResourceResponse _mockLiferayResourceResponse = new MockLiferayResourceRequest()), with that the results always were same as the CI.

image

Actions:
To surpass this, I'm cleaning the objects after each test run, and also setting as null right after the assert.

### CODE REVIEW

**Comment by rijema** [comment]
ci:test:sf

**Comment by rijema** [comment]
As this ticket is a flaky one, related to CI (mainly the acceptance routine), I'll try to see the results from relevant/objects pointing to this test.

**Comment by rijema** [comment]
ci:test:relevant

**Comment by rijema** [comment]
ci:reevaluate:1277401_12792

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/web/internal/BaseExportImportTestCase.java [hunk 1]
```diff
@@ -143,25 +155,26 @@ protected void testExportImportJSON(
...
 
 		JSONAssert.assertEquals(
 			expectedJSON,
 			String.valueOf(
-				mockLiferayResourceResponse.getPortletOutputStream()),
+				_mockLiferayResourceResponse.getPortletOutputStream()),
 			JSONCompareMode.LENIENT);
+
+		_mockLiferayResourceRequest = null;
+		_mockLiferayResourceResponse = null;
```
[the remark this answers is not in the document]
Comment (carolmariaabb) [reply modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/web/internal/BaseExportImportTestCase.java:177]: My suspicious is that the [getId](https://github.com/liferay/liferay-portal/blob/80a3903d056fbcf1c0b403362df67d9c4b8d8fb4/modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/web/internal/BaseExportImportTestCase.java#L150) method is returning the wrong Object Folder [here](https://github.com/liferay/liferay-portal/blob/80a3903d056fbcf1c0b403362df67d9c4b8d8fb4/modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/web/internal/object/definitions/portlet/action/test/ObjectFolderExportImportTest.java#L147).

We are making a search by the object folder name [here](https://github.com/liferay/liferay-portal/blob/80a3903d056fbcf1c0b403362df67d9c4b8d8fb4/modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/web/internal/object/definitions/portlet/action/test/ObjectFolderExportImportTest.java#L245-L250).

Could you check if there is any other test that could be creating a folder with the same name? This test may be running in parallel, impacting our test.

**Review by carolmariaabb — CHANGES_REQUESTED** [review]

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/web/internal/BaseExportImportTestCase.java
[the remark this answers is not in the document]
Comment (rijema) [reply modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/web/internal/BaseExportImportTestCase.java:1]: Hi. Right after mocking the request and response as setting as null.

Just my thought: Even each test is creating a instance of the objects, and if the same general test (testExportImportObjectFolder) in another environment, I was thinking if the JVM could point to an already existing object by mistake once in a while when dealing with several threads that is executing a similar queue.

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/web/internal/BaseExportImportTestCase.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (rijema) [reply modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/web/internal/BaseExportImportTestCase.java:177]: Yes, there is. On SiteInitializer, the developers recently created ObjectFolders with the same names and ERC. Considering testray really relies over parallel actions, would make sense to isolate our test case using the @Transactional annotation?

**Comment by carolmariaabb** [comment]
I'll close this PR for now. Let's discuss a new approach later.

**Pull request closed without merging**